---
title: "Function"
author: "Alice"
date: "2022/2/12"
output: html_document
---

```{r}
check <- function(table, gene.col, gt.col, flag = c("Alleles", "Snps", "Carrier")){
```

```{r}
library(magrittr)
library(stringr)
table <- tibble::tibble(CHROM=1, 
                POS=2:7, 
                ID = ".", 
                gene.col=c("g1", "g1", "g2", "g3", "g4","g4"), 
                gt.col=c("0/1", "1/0", rep("0/0", 4)),
                s2=c("1/0", "0/0", "1/0", "0/0", "0/1", "./."),
                s3=c("0/0", "1/1", "0/0","0/0", "./.", "./."),
                s4=c(rep("0/0", 3), "0/1", "1/1", "0/1"))
 


gt_to_num <- function(string) {
  sum(as.numeric(str_extract_all(string, "\\d")[[1]]))
}

table %>%
  dplyr::rowwise() %>%
  dplyr::mutate(across(gt.col:ncol(table), gt_to_num))
print(table)

  Allele <- table %>% 
    dplyr::group_by(gene.col) %>%
    dplyr::summarise(across((gt.col-1):(ncol(table)-1),sum))
  
  carrier <- Allele
  if(carrier[row,col] > 1){
    carrier[row, col] = 1
  }
  
  if (flag == "Allele") {
    return(Allele)
  }
  if (flag == "snps") {
    return(snps)
  } 
  if(flag == "carrier"){
    return(carrier)
  }
}
```
```{r}
dt <- tibble(gene=c("g1","g2","g3","g4"),
             s1=c(1,1,0,0),
             s2=s1,
             s3=c(1,1,1,1),
             s4=s3)
gs <- c("g2","g3")

Task2 <- library(tidyverse)
library(dplyr)

count_gene_set_variant_refined <- function(Table, GeneSet, Name) {
  
  Set <- read_lines(GeneSet)
  
 
  variant.per.gene_set.per.sample<- Table %>%
    filter(gene %in% Set) %>%
    summarise(across(2:ncol(Table),sum)) %>%
    pivot_longer( everything(), names_to = "sampleID", values_to = Name)

  return(variant.per.gene_set.per.sample)
}
```


```{r}
library(tidyverse)

input <- tibble(CHROM=1, 
                POS=2:7, 
                ID = ".", 
                GENE=c("g1", "g1", "g2", "g3", "g4","g4"), 
                s1=c("0/1", "1/0", rep("0/0", 4)),
                s2=c("1/0", "0/0", "1/0", "0/0", "0/1", "./."),
                s3=c("0/0", "1/1", "0/0","0/0", "./.", "./."),
                s4=c(rep("0/0", 3), "0/1", "1/1", "0/1"))


gt2count <- function(genotype){
  str_extract_all(genotype, "\\d") %>%
    map_int(~sum(as.integer(.)))
}

input %>%
  select(5:8) %>%
  map_df(gt2count )
  

```
  
  